source/text/labelgeometry.ts

```diff
@@ -102,65 +140,159 @@ export class LabelGeometry extends Geometry {
      */
     protected unbindBuffers(indices: Array<GLuint>): void {
         /* Please note the implicit unbind in attribEnable is skipped */
-        const l = this._buffers.length;
-        for (let i = 0; i < l; i++) {
-            this._buffers[i].attribDisable(indices[i], true, true);
-        }
-    }
 
-    /**
-     * Specifies/invokes the draw of all labels.
-     */
-    @Initializable.assert_initialized()
-    draw(): void {
-        this.context.gl2facade.drawArraysInstanced(this.context.gl.TRIANGLE_STRIP, 0, 4, this._glyphCount);
+        this._vertices.attribDisable(this._vertexLocation, false, false);
+        this._texCoords.attribDisable(this._texCoordLocation, false, false);
+
+        this._origins.attribDisable(this._originLocation, false, false);
+
+        this._tangents.attribDisable(this._tangentLocation, false, false);
+        this._bitangents.attribDisable(this._bitangentLocation, false, false);
     }
 
+
     /**
      * Creates the vertex buffer object (VBO) and creates and initializes the buffer's data store.
-     * @param aQuadVertex - Attribute binding point for vertices.
-     * @param aTexCoord - Attribute binding point for texture coordinates.
-     * @param aOrigin - Attribute binding point for glyph origin coordinates
-     * @param aTangent - Attribute binding point for glyph tangent coordinates.
-     * @param aUp - Attribute binding point for glyph up-vector coordinates.
+     * @param vertexLocation - Attribute binding point for vertices.
+     * @param texCoordLocation - Attribute binding point for texture coordinates.
+     * @param originLocation - Attribute binding point for glyph origin coordinates
+     * @param tangentLocation - Attribute binding point for glyph tangent coordinates.
+     * @param bitangentLocation - Attribute binding point for glyph up-vector/bitangent coordinates.
      */
-    initialize(aQuadVertex: GLuint, aTexCoord: GLuint, aOrigin: GLuint, aTangent: GLuint, aUp: GLuint): boolean {
+    initialize(
+        vertexLocation: GLuint = 0,
+        texCoordLocation: GLuint = 1,
+        originLocation: GLuint = 2,
+        tangentLocation: GLuint = 3,
+        bitangentLocation: GLuint = 4): boolean {
+
+        this._vertexLocation = vertexLocation;
+        this._texCoordLocation = texCoordLocation;
+        this._originLocation = originLocation;
+        this._tangentLocation = tangentLocation;
+        this._bitangentLocation = bitangentLocation;
+
         const gl = this.context.gl;
         const valid = super.initialize(
-            [gl.ARRAY_BUFFER, gl.ARRAY_BUFFER, gl.ARRAY_BUFFER, gl.ARRAY_BUFFER, gl.ARRAY_BUFFER],
-            [aQuadVertex, aTexCoord, aOrigin, aTangent, aUp]);
+            [gl.ARRAY_BUFFER, gl.ARRAY_BUFFER, gl.ARRAY_BUFFER, gl.ARRAY_BUFFER, gl.ARRAY_BUFFER]);
 
         /**
          * These vertices are equal for all quads. There actual position will be changed using
          * origin, tangent and up(-vector).
```

Comment:
up-vector is now called bitangent (although I don't like it to be bitangent ^^ but we should be consistent in the docs...)